src/mito2/src/sst/index/fulltext_index/creator.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+use common_telemetry::warn;
+use datatypes::schema::FulltextAnalyzer;
+use index::fulltext_index::create::{FulltextIndexCreator, TantivyFulltextIndexCreator};
+use index::fulltext_index::{Analyzer, Config};
+use puffin::blob_metadata::CompressionCodec;
+use puffin::puffin_manager::{PuffinWriter, PutOptions};
+use snafu::{ensure, ResultExt};
+use store_api::metadata::RegionMetadataRef;
+use store_api::storage::{ColumnId, ConcreteDataType, RegionId};
+
+use crate::error::{
+    CastVectorSnafu, CreateFulltextCreatorSnafu, FieldTypeMismatchSnafu, FulltextFinishSnafu,
+    FulltextOptionsSnafu, FulltextPushTextSnafu, OperateAbortedIndexSnafu, PuffinAddBlobSnafu,
+    Result,
+};
+use crate::read::Batch;
+use crate::sst::file::FileId;
+use crate::sst::index::fulltext_index::INDEX_BLOB_TYPE;
+use crate::sst::index::intermediate::IntermediateManager;
+use crate::sst::index::puffin_manager::SstPuffinWriter;
+use crate::sst::index::statistics::{ByteCount, RowCount, Statistics};
+
+/// `SstIndexCreator` is responsible for creating fulltext indexes for SST files.
+pub struct SstIndexCreator {
+    /// Creators for each column.
+    creators: HashMap<ColumnId, SingleCreator>,
+    /// Whether the index creation was aborted.
+    aborted: bool,
+    /// Statistics of index creation.
+    stats: Statistics,
+}
+
+impl SstIndexCreator {
+    /// Creates a new `SstIndexCreator`.
+    pub async fn new(
+        region_id: &RegionId,
+        sst_file_id: &FileId,
+        intermediate_manager: &IntermediateManager,
+        metadata: &RegionMetadataRef,
+        compress: bool,
+        mem_limit: usize,
+    ) -> Result<Self> {
+        let mut creators = HashMap::new();
+
+        for column in &metadata.column_metadatas {
+            let options =
+                column
+                    .column_schema
+                    .fulltext_options()
+                    .context(FulltextOptionsSnafu {
+                        column_name: &column.column_schema.name,
+                    })?;
+
+            // Relax the type constraint here as many types can be casted to string.
```

Comment:
I think we can constrain the type to be string, other types such as DateTime or integers, even though they can be cast into string, but the full text on for them is useless.

Reply:
My idea is that specific constraints can be implemented by the frontend, and by relaxing constraints in the mito, many things can function.